### ISSUE TITLE
Loss drops to 0 and nce_samples becomes 1 during training on Amazon Beauty and Toys datasets

### ISSUE THREAD

**Issue by xhd0728** [bytedance/HLLM#33]
Hi!

While reproducing experiments using the Amazon Beauty and Toys datasets with TinyLlama-1.1B for both User LLM and Item LLM, I encountered abnormal training behavior after a certain step. Specifically:

## Issue Details:
- The loss abruptly drops to 0.
- The `nce_samples` value drops to 1.
- This issue persists for the remainder of the training process.

## Training Script:

```bash
#!/bin/bash
start_time="$(date +%FT%T)"

cd code && python3 main.py \
    --config_file overall/LLM_deepspeed.yaml HLLM/HLLM.yaml \
    --MAX_ITEM_LIST_LENGTH 50 \
    --epochs 5 \
    --optim_args.learning_rate 1e-4 \
    --checkpoint_dir checkpoint/toys \
    --loss nce \
    --MAX_TEXT_LENGTH 256 \
    --dataset amazon_toys \
    --train_batch_size 1 \
    --text_path information \
    --item_pretrain_dir ../model/TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T \
    --user_pretrain_dir ../model/TinyLlama/TinyLlama-1.1B-intermediate-step-1431k-3T \
    --stage 3 \
    --num_negatives 512 \
    --gradient_checkpointing True \
    2>&1 | tee logs/hllm-toys-${start_time}.log
```

## Relevant Logs:

```
...
Train [  0/  5]:  34%|███▍      | 4560/13429 [3:55:52<7:24:33,  3.01s/it, lr: 0.0000676 loss: 4.6875 data: 0.033 fwd: 1.077 bwd: 1.850 nce_samples: 514.000 nce_top1_acc: 0.167 nce_top5_acc: 0.500 nce_top10_acc: 0.500 nce_top50_acc: 0.500 nce_top100_acc: 0.833]
Train [  0/  5]:  34%|███▍      | 4600/13429 [3:57:53<7:22:30,  3.01s/it, lr: 0.0000682 loss: 0.0000 data: 0.006 fwd: 1.195 bwd: 1.826 nce_samples: 1.000 nce_top1_acc: 1.000 nce_top5_acc: 1.000 nce_top10_acc: 1.000 nce_top50_acc: 1.000 nce_top100_acc: 1.000]
...
```

## Expected Behavior:

- The loss should not prematurely drop to 0 during training.
- The number of `nce_samples` should remain consistent with task requirements instead of collapsing to 1.

I confirm that I have not modified any other files in the project. Could you help identify potential causes for this behavior? Are there specific debugging steps or parameter adjustments you would recommend?

Thank you! Please let me know if additional information or logs are needed.
